pdfminer/encodingdb.py

```diff
@@ -1,28 +1,56 @@
 
 import re
-from .psparser import PSLiteral
+
+import six  # Python 2+3 compatibility
+
 from .glyphlist import glyphname2unicode
 from .latin_enc import ENCODING
+from .psparser import PSLiteral
 
-import six # Python 2+3 compatibility
-
-STRIP_NAME = re.compile(r'[0-9]+')
+HEXADECIMAL = re.compile(r'[0-9a-fA-F]+')
 
 
-##  name2unicode
-##
 def name2unicode(name):
-    """Converts Adobe glyph names to Unicode numbers."""
-    if name in glyphname2unicode:
-        return glyphname2unicode[name]
-    m = STRIP_NAME.search(name)
-    if not m:
-        raise KeyError(name)
-    return six.unichr(int(m.group(0)))
+    """Converts Adobe glyph names to Unicode numbers.
```

Comment:
This looks good. I am wondering whether it would make sense to have a try except block inside the function. I happened to see some logging in other files, so maybe we can define a logger in this file, put a try/except inside the function and log any excpetion found using `logging.exception(ERROR MESSAGE")`, then `raise KeyError`? That way we should be able to propagate the error message, the stacktrace and also raise keyerror. Not sure how relevant it is here, though.

Reply:
I agree that we could be more verbose about what's going on. So I've added a message to each `KeyError` that is raised inside `name2unicode()` and use this message outside `name2unicode()` to log a debug statement. 

I choose to made it a debug statement because the `KeyError`'s are not really exceptional when parsing pdf fonts. I think that's the reason why `KeyError`'s are used instead of returning an empty string. For some usages of `name2unicode()` it is allowed to have glyph names that do not match (e.g. during font parsing). In other cases it is not allowed (e.g. in `FileUnicodeMap.add_cid2unichr()`). Not sure if this difference was intended, but I interpret it this way.